### PR TITLE
Implement slack context provider

### PR DESCRIPTION
**Pull request by hitesh-1997**
- Implements Slack context provider for OpenCtx, which allows to search for relevant threads when mentioning a slack channel and use all the messages in the thread as the context to AI.
- Uses [@slack/web-api](https://www.npmjs.com/package/@slack/web-api) to fetch all data related to slack such as channel list, use slack search api and get thread messages etc. 
- Requires authorization using a personal User API token created via slack app.

## Test Plan
- Tested using a personal user auth token created via slack app.

## ToDo
1. Add a local setup to handle the `https` redirects from slack to get the user auth token.
2. For now, we use a common user access token, with the limitation that we can only access public channels and not private channels.

## Demo

https://github.com/sourcegraph/openctx/assets/20701220/15b6096f-ff1c-42dd-ba66-961fa7cbeebe

### CODE REVIEW

File: bin/config.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "enable": true,
+  "debug": true,
+  "providers": {
+    "file:///Users/hiteshsagtani/dev/openctx/provider/slack-channel-context/dist/bundle.js": true
```
Comment (toolmantim) [inline bin/config.json:5]: Accidental git add?

Comment (hitesh-1997) [reply bin/config.json:5]: oh damn, yes pushed it by mistake, thanks for noticing !!

File: provider/slack-channel-context/index.ts [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+            return undefined
+        })
+        const allMessages = await Promise.all(allPromises)
+        const allSlackContext = []
+        for (const message of allMessages) {
+            if (message) {
+                allSlackContext.push(message)
+            }
+        }
+        return allSlackContext
```
Comment (valerybugakov) [inline provider/slack-channel-context/index.ts:211]:
```suggestion
        return allMessages.filter(Boolean)
```

File: provider/slack-channel-context/package.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "@openctx/provider-slack-channel-context",
+  "version": "0.0.8",
```
Comment (valerybugakov) [inline provider/slack-channel-context/package.json:3]:
```suggestion
  "version": "0.0.1",
```

**Review by valerybugakov — APPROVED** [review]

File: provider/slack-channel-context/index.ts [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+function getSlackClient(settingsInput: Settings) {
+    if (settingsInput.slackAuthToken) {
+        const slackWebClient = new WebClient(settingsInput.slackAuthToken)
+        return slackWebClient
+    }
+    throw new Error('must provide a Slack user auth token in the `slackAuthToken` user settings field')
+}
+
+async function listAllChannels(client: WebClient): Promise<ChannelInfo[]> {
+    let allChannels: ChannelInfo[] = []
```
[the remark this answers is not in the document]
Comment (hitesh-1997) [reply provider/slack-channel-context/index.ts:30]: we modify this value as we get more channel information `allChannels = [...allChannels, ...channelInfo]`, so used `let` here

**Pull request merged**